Source/Async.swift

```diff
@@ -0,0 +1,177 @@
+    ///
+    /// - Returns: The `DataTask`.
+    public func string(dataPreprocessor: DataPreprocessor = StringResponseSerializer.defaultDataPreprocessor,
+                       encoding: String.Encoding? = nil,
+                       emptyResponseCodes: Set<Int> = StringResponseSerializer.defaultEmptyResponseCodes,
+                       emptyRequestMethods: Set<HTTPMethod> = StringResponseSerializer.defaultEmptyRequestMethods) -> DataTask<String> {
+        serialize(using: StringResponseSerializer(dataPreprocessor: dataPreprocessor,
+                                                  encoding: encoding,
+                                                  emptyResponseCodes: emptyResponseCodes,
+                                                  emptyRequestMethods: emptyRequestMethods))
+    }
+
+    /// Creates a `DataTask` to `await` serialization of a `Data` value.
+    ///
+    /// - Parameters:
+    ///   - dataPreprocessor:    `DataPreprocessor` used to prepare the received `Data` for serialization.
+    ///   - emptyResponseCodes:  The HTTP response codes for which empty responses are allowed. `[204, 205]` by default.
+    ///   - emptyRequestMethods: The HTTP request methods for which empty responses are allowed. `[.head]` by default.
+    ///
+    /// - Returns: The `DataTask`.
+    public func data(dataPreprocessor: DataPreprocessor = DataResponseSerializer.defaultDataPreprocessor,
+                     emptyResponseCodes: Set<Int> = DataResponseSerializer.defaultEmptyResponseCodes,
+                     emptyRequestMethods: Set<HTTPMethod> = DataResponseSerializer.defaultEmptyRequestMethods) -> DataTask<Data> {
+        serialize(using: DataResponseSerializer(dataPreprocessor: dataPreprocessor,
+                                                emptyResponseCodes: emptyResponseCodes,
+                                                emptyRequestMethods: emptyRequestMethods))
+    }
+
+    /// Creates a `DataTask` to `await` serialization using the provided `DataResponseSerializerProtocol` instance.
+    ///
+    /// - Parameters:
+    ///    - serializer: Response serializer responsible for serializing the request, response, and data.
+    ///
+    /// - Returns: The `DataTask`.
+    public func serialize<Serializer: DataResponseSerializerProtocol>(using serializer: Serializer) -> DataTask<Serializer.SerializedObject> {
+        dataTask {
+            self.response(queue: .asyncCompletionQueue,
+                          responseSerializer: serializer,
+                          completionHandler: $0)
+        }
+    }
+
+    private func dataTask<Value>(forResponse onResponse: @escaping (@escaping (DataResponse<Value, AFError>) -> Void) -> Void) -> DataTask<Value> {
+        let handle = async {
+            await withTaskCancellationHandler {
+                self.cancel()
+            } operation: {
+                await withCheckedContinuation { continuation in
```

Comment:
I think that using `withUnsafeContinuation` will be beter for performance.

Reply:
Not in any noticeable fashion since we're waiting for a network response and all of Alamofire's other internal machinery to complete anyway. Generally Swift prefers the safe operations (checked arithmetic, bounds-checked collections, etc.) to unsafe unless there's critical performance to be reclaimed, which isn't the case here.